pkg/cmd/roachtest/tests/multiregion_leasing.go

```diff
@@ -0,0 +1,141 @@
+				}
+				conn := c.Conn(ctx, t.L(), c.Node(1)[0])
+				for s, v := range options {
+					if _, err := conn.Exec(fmt.Sprintf("SET CLUSTER SETTING %s='%s'", s, v)); err != nil {
+						t.Fatal(err)
+					}
+				}
+				conn.Close()
+				c.Stop(ctx, t.L(), option.DefaultStopOpts(), c.All())
+			}
+
+			c.Start(ctx, t.L(), option.DefaultStartOpts(), defaultOpts, c.All())
+			// Create 600 tables inside the database, which is above our default lease
+			// refresh limit of 500. This will only be done when session based leasing
+			// is enabled.
+			numTables := 600 / len(c.All())
+			grp := ctxgroup.WithContext(ctx)
+			if modeIdx == 0 {
+				for _, n := range c.All() {
+					var node = n
+					grp.GoCtx(func(ctx context.Context) error {
+						conn, err := c.ConnE(ctx, t.L(), node)
+						if err != nil {
+							return err
+						}
+						defer conn.Close()
+						tx, err := conn.Begin()
+						if err != nil {
+							return err
+						}
+						for i := 0; i < numTables; i++ {
+							createTbl := fmt.Sprintf("CREATE TABLE table_%d_%d (i int);", node, i)
+							_, err := tx.Exec(createTbl)
+							if err != nil {
+								return err
+							}
+						}
+						return tx.Commit()
+					})
+				}
+				if err := grp.Wait(); err != nil {
+					t.Fatal(err)
+				}
+			}
+			// Next spawn a thread on each node to select from all the tables created
+			// above. We are going to do two passes, with multiple threads. After 30
+			// seconds the lease will expire in the expiry based model, and checking
+			// all of these tables will take more than the expiry time. Additionally,
+			// a 100 tables will *never* be auto-refreshed.
+			grp = ctxgroup.WithContext(ctx)
+			for _, n := range c.All() {
+				var node = n
+				const numWorkers = 8
+				tablesPerWorker := numTables / numWorkers
+				for workerId := 0; workerId < numWorkers; workerId++ {
+					workerId := workerId
+					grp.GoCtx(func(ctx context.Context) error {
+						conn, err := c.ConnE(ctx, t.L(), node)
+						if err != nil {
+							return err
+						}
+						defer conn.Close()
+						for iter := 0; iter < 8; iter++ {
+							for i := workerId * tablesPerWorker; i < (workerId+1)*tablesPerWorker; i++ {
+								for j := 1; j <= len(c.All()); j++ {
+									createTbl := fmt.Sprintf("SELECT * FROM table_%d_%d;", j, i)
+									_, err := conn.Exec(createTbl)
+									if err != nil {
+										return err
+									}
+								}
+							}
+						}
+						return nil
+					})
+				}
+			}
+			t.Status("Benchmarking Session Based Leasing: ", sessionBasedLeasingEnabled)
+			startTime := timeutil.Now()
+			if err := grp.Wait(); err != nil {
+				t.Fatal(err)
+			}
+			c.Stop(ctx, t.L(), option.DefaultStopOpts())
+			durations[modeIdx] = timeutil.Since(startTime)
+		}()
+	}
+
+	// Confirm that session based leasing took less time for the selects.
+	if durations[0] > durations[1] {
+		t.Fatal("Expected session based leasing to be faster with many descriptors.")
```

Comment:
For your own sanity when this fails, it's probably better to log the durations. I don't see them logged anywhere.